azure-iot-device/tests/iothub/aio/test_async_clients.py

```diff
@@ -2266,6 +2303,50 @@ def fail_send_output_event(message, callback):
         assert isinstance(sent_message, Message)
         assert sent_message.data == message_input
 
+    @pytest.mark.it("Raises error when message data size is greater than 256 KB")
+    async def test_raises_error_when_message_to_output_data_greater_than_256(
+        self, client, iothub_pipeline
+    ):
+        output_name = "some_output"
+        data_input = "serpensortia" * 256000
+        message = Message(data_input)
+        with pytest.raises(ValueError) as e_info:
+            await client.send_message_to_output(message, output_name)
+        assert "256 KB" in e_info.value.args[0]
+        assert iothub_pipeline.send_output_event.call_count == 0
+
+    @pytest.mark.it("Raises error when message size is greater than 256 KB")
+    async def test_raises_error_when_message_to_output_size_greater_than_256(
+        self, client, iothub_pipeline
+    ):
+        output_name = "some_output"
+        data_input = "serpensortia"
+        message = Message(data_input)
+        message.custom_properties["spell"] = data_input * 256000
+        with pytest.raises(ValueError) as e_info:
+            await client.send_message_to_output(message, output_name)
+        assert "256 KB" in e_info.value.args[0]
+        assert iothub_pipeline.send_output_event.call_count == 0
+
+    @pytest.mark.it("Does not raises error when message data size is equal to 256 KB")
+    async def test_raises_error_when_message_to_output_data_equal_to_256(
```

Comment:
>test_raises_error_when_message_to_output_data_equal_to_256 [](start = 14, length = 58)

I go back and forth on the utility of the equal_to_256 tests.  I don't think I would be sad to see them disappear.

Reply:
I want to keep it to check the boundary..but right now it is not serving any purpose. but i do want to address it after i do the content type and content encoding as things are bound to change then.

---
In reply to: [374956555](https://github.com/Azure/azure-iot-sdk-python/pull/447#discussion_r374956555) [](ancestors = 374956555)